drop-boxes/register-pacbio-dropbox/register-pacbio-dropbox.py

```diff
@@ -0,0 +1,166 @@
+
+See also https://pacbiofileformats.readthedocs.io/en/10.0/DataSet.html for details
+
+Note:
+print statements go to: ~openbis/servers/datastore_server/log/startup_log.txt
+'''
+import sys
+sys.path.append('/home-link/qeana10/bin/')
+
+import checksum
+import re
+import time
+import os
+import ch.systemsx.cisd.etlserver.registrator.api.v2
+from java.io import File
+from org.apache.commons.io import FileUtils
+from ch.systemsx.cisd.openbis.generic.shared.api.v1.dto import SearchCriteria
+from ch.systemsx.cisd.openbis.generic.shared.api.v1.dto import SearchSubCriteria
+
+# expected:
+# *Q[Project Code]^4[Sample No.]^3[Sample Type][Checksum]*.*
+pattern = re.compile('Q\w{4}[0-9]{3}[a-zA-Z]\w')
+
+######## Sample Tracking related import
+from life.qbic.sampletracking import SampleTracker
+from life.qbic.sampletracking import ServiceCredentials
+from java.net import URL
+
+import sample_tracking_helper_qbic as tracking_helper
+#### Setup Sample Tracking service
+SERVICE_CREDENTIALS = ServiceCredentials()
+SERVICE_CREDENTIALS.user = tracking_helper.get_service_user()
+SERVICE_CREDENTIALS.password = tracking_helper.get_service_password()
+SERVICE_REGISTRY_URL = URL(tracking_helper.get_service_reg_url())
+DATA_AVAILABLE_JSON = tracking_helper.get_data_available_status_json()
+
+### We need this object to update the sample status later
+SAMPLE_TRACKER = SampleTracker.createLocationIndependentSampleTracker(SERVICE_REGISTRY_URL, SERVICE_CREDENTIALS)
+
+DATASET_TYPE = "Q_NGS_PACBIO_DATA"
+SAMPLE_TYPE = "Q_NGS_PACBIO_RUN"
+EXPERIMENT_TYPE = "Q_NGS_PACBIO_MEASUREMENT"
+
+def isExpected(identifier):
+        try:
+                id = identifier[0:9]
+                #also checks for old checksums with lower case letters
+                return checksum.checksum(id)==identifier[9]
+        except:
+                return False
+
+def process(transaction):
+        context = transaction.getRegistrationContext().getPersistentMap()
+
+        # Get the incoming path of the transaction
+        incomingPath = transaction.getIncoming().getAbsolutePath()
+
+        key = context.get("RETRY_COUNT")
+        if (key == None):
+                key = 1
+
+
+        # Get the name of the incoming file
+        name = transaction.getIncoming().getName()
+        
+        identifier = pattern.findall(name)[0]
+        #identifier = name
+        if isExpected(identifier):
+                project = identifier[:5]
+        else:
+                print "The identifier "+identifier+" did not match the pattern Q[A-Z]{4}\d{3}\w{2} or checksum"
+        
+        # clean unnecessary metadata created by the datahandler
+        for f in os.listdir(incomingPath):
+            if ".testorig" in f:
+                os.remove(os.path.realpath(os.path.join(incomingPath,f)))
+        # check if expected files are all there
+        bam_index_ext = ".bam.pbi" # easier than implementing handling of multiple suffixes atm
+        mandatory_extensions = [".xml", ".bam", bam_index_ext]
```

Comment:
In theory it could also be possible to read the xml file and check if the files are provided as specified in the xml format.

Reply:
Something like this will come with the next version for sure 👍